pkg/controller/controller.go

```diff
@@ -117,16 +118,33 @@ func computeChangeList(current []model.Config, desired []model.Config, errHandle
 	return changeList
 }
 
-// getErrHandler returns a error handler func that re-adds the athenz domain back to queue
+// getCallbackHandler returns a error handler func that re-adds the athenz domain back to queue
 // this explicit func definition takes in the key to avoid data race while accessing key
-func (c *Controller) getErrHandler(key string) processor.OnErrorFunc {
+func (c *Controller) getCallbackHandler(key string) processor.OnCompleteFunc {
 	return func(err error, item *processor.Item) error {
-		if err != nil {
-			if item != nil {
-				log.Errorf("Error performing %s on %s: %s", item.Operation, item.Resource.Key(), err.Error())
-			}
-			c.queue.AddRateLimited(key)
+
+		if err == nil {
+			return nil
+		}
+		if item != nil {
+			log.Errorf("Error performing %s on %s: %s", item.Operation, item.Resource.Key(), err.Error())
+		}
+		if apiErrors.IsNotFound(err) || apiErrors.IsAlreadyExists(err) {
+			log.Infof("Error is non-retryable %s", err)
+			return nil
```

Comment:
Should we not add to the queue here?

Reply:
This is a case, where say a CREATE got scheduled, but another CREATE created the resource and API returned AlreadyExists. Now, the change for the CREATE would trigger another sync that should take care of further changes if any needed, so we wouldn't have to add it back to the queue in this case.

Reply:
Ah got it, that sounds good.